Modules/_ctypes/cfield.c

```diff
@@ -1289,7 +1289,9 @@ s_set(void *ptr, PyObject *value, Py_ssize_t length)
     }
 
     data = PyBytes_AS_STRING(value);
-    size = strlen(data); /* XXX Why not Py_SIZE(value)? */
+    /* bpo-39593: XXX Why not Py_SIZE(value)? */
```

Comment:
I don't think that it's correct to add a test which indirectly specify the behavior and keep this comment which suggests to change the behavior.

I suggest to rewrite the comment to explain that we do use strlen() on purpose, rather than PyBytes_GET_SIZE() since we truncate characters after a null character **on purpose**. Keep the bpo number in the comment.

Reply:
Thanks, got it. I updated the description info, do you think it's clear enough?